Client/Frontend/Browser/User Scripts/FarblingProtectionHelper.swift

```diff
@@ -0,0 +1,127 @@
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+/// A class that helps in creating farbling data
+class FarblingProtectionHelper {
+  /// Represents `JSON` data that needs to be passed to `FarblingProtection.js`
+  struct FarblingData: Encodable {
+    /// Represents the `JSON` data that is needed to construct a fake WebKit `Plugin`
+    struct FakePluginData: Encodable {
+      let name: String
+      let filename: String
+      let description: String
+      let mimeTypes: [FakeMimeTypeData]
+    }
+
+    /// Represents the `JSON` data that is needed to construct a fake WebKit `MimeType`
+    struct FakeMimeTypeData: Encodable {
+      let suffixes: String
+      let type: String
+      let description: String
+    }
+
+    /// A value between 0.99 and 1 to fudge audio data
+    ///
+    /// A value between 0.99 to 1 means the values in the destination will
+    /// always be within the expected range of -1 and 1.
+    /// This small decrease should not affect affect legitimite users of this api.
+    /// But will affect fingerprinters by introducing a small random change.
+    let fudgeFactor: Float
+    /// A value representing a fake voice name that will be used to add a fake voice
+    let fakeVoiceName: String
+    /// Fake data that is to be used to construct fake plugins
+    let fakePluginData: [FakePluginData]
+    /// This value is used to get a random index between 0 and an unknown count
+    ///
+    /// It's important to have a value between 0 - 1 in order to be within the array bounds
+    let randomVoiceIndexScale: Float
+  }
+
+  /// Variables representing the prefix of a randomly generated strings used as the plugin name
+  private static let pluginNameFirstParts: [String?] = [
+    "Chrome", "Chromium", "Brave", "Web", "Browser",
+    "OpenSource", "Online", "JavaScript", "WebKit",
+    "Web-Kit", "WK", nil
+  ]
+
+  /// Variables representing the middle of a randomly generated strings used as the plugin name
+  private static let pluginNameSecondParts: [String?] = [
+    "PDF", "Portable Document Format",
+    "portable-document-format", "document", "doc",
+    "PDF and PS", "com.adobe.pdf", nil
+  ]
+
+  /// Variables representing the suffix of a randomly generated strings used as the plugin name
+  private static let pluginNameThirdParts: [String?] = [
+    "Viewer", "Renderer", "Display", "Plugin",
+    "plug-in", "plug in", "extension", nil
+  ]
+
+  /// A list of fake voice names to be used to generate a fake `SpeechSynthesizer` voice
+  private static let fakeVoiceNames: [String] = [
+    "Hubert", "Vernon", "Rudolph", "Clayton", "Irving",
+    "Wilson", "Alva", "Harley", "Beauregard", "Cleveland",
+    "Cecil", "Reuben", "Sylvester", "Jasper"
+  ]
+
+  static func makeFarblingParams(from randomConfiguration: RandomConfiguration) throws -> String {
+    var generator = ARC4RandomNumberGenerator(seed: randomConfiguration.domainKeyData.getBytes())
```

Comment:
Is this random number generator supposed to be secure? Later on in the code, the farbling data is using `HMAC SHA-256`, but this random number generator is using `arc4random`. So should this generator be cryptographically secure?

Reply:
I don't know. I borrowed it from apple's Tensor flow code. I'd be happy to discuss this however.
Though it's not later on in the code where I use SHA-256. the seed of this generator is the result of the HMAC SHA-256 encoding.

Reply:
@thypon and @Brandon-T I dropped the ARC4 implementation in favour of manually doing an `srand48` and `drand48`. Both are standard seed and get rand functions part of iOS (no need to include any 3rd party implementations)